File: learn-java/src/main/java/boki/learnjava/flux/FluxEx4.java
package boki.learnjava.flux;

import lombok.extern.slf4j.Slf4j;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Slf4j
public class FluxEx4 {
    public static void main(String[] args) {
        Flux.concat(
                Flux.just("Venus"),
                Mono.just("Earth"),
                Flux.just("Mars"))
            .collectList() // 주석하고 실행하면 각각의 데이터들이 전달됨
            .subscribe(planetList -> log.info("# Solar System: {}", planetList));
    }
}

File: learn-kt/src/main/kotlin/boki/learn/flux/FluxEx4.kt
package boki.learn.flux

import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

fun main() {
    Flux.concat<Any>(
        Flux.just("Venus"),
        Mono.just("Earth"),
        Flux.just("Mars")
    )
        .collectList() // 주석하고 실행하면 각각의 데이터들이 전달됨
        .subscribe { println("# Solar System: $it") }
}